binary_transparency/firmware/api/firmware_metadata.go

```diff
@@ -0,0 +1,68 @@
+package api
+
+// FirmwareMetadata represents a firmware image and related info.
+type FirmwareMetadata struct {
+	// Raw is the bytestream this object was parsed from.
+	Raw []byte
+	// Signature is the bytestream of the signature over the Raw field above.
+	Signature []byte
+
+	////// What's this firmware for? //////
+
+	// TODO: possibly want a vendor/device/publisher model?
+
+	// DeviceName is a human readable description of the target device.
+	DeviceName string
+
+	// DeviceID is a unique machine-readable ID for the target device.
+	// Monitors should check that DeviceName and DeviceID correlate.
+	DeviceID uint64
+
+	////// Who published it //////
+
+	// FirmwarePublisher is a human-readable desciption of the publisher.
+	FirmwarePublisher string
+
+	// FirmwarePublickeyHash is a hint for signing identity.
+	// Monitors should check that this is indeed the key which signed
+	// this metadata, *and* that the FirmwarePublisher string matches the key's
+	// identity.
+	FirmwarePublickeyHash []byte
+
+	////// What's its identity? //////
+
+	// FirmwareType identifies where it runs within the device.
+	FirmwareType string
+
+	// FirmwareRevision specifies which version of firmware this is.
+	// TODO: string?
```

Comment:
Here and in other TODOs. Please use TODO(<username>|<github_issue>): An actionable task (and conditions). i.e. "Change to use a freeform or <major>.<minor> format string, once we decide which one is best suited."

Reply:
That sounds like a great idea in general, but maybe a bit heavyweight just at the moment given this is really just placeholder/breaking ground?

Reply:
OK, maybe not bugs, at least change to TODO(alcutter)? (I suggested bugs to make things more generic, but agreed it might be a bit heavyweight.

And please make the TODOs actionable tasks. (Yep, even if this is placeholder! please? :) )

Reply:
todone